### DISCUSSION TITLE
examples：HelloWorld.cpp compile error

### DISCUSSION THREAD

**Discussion by zhlongfj** [uNetworking/uWebSockets#1061] [Q&A] [answered]
uWS::SSLApp({
      .key_file_name = "../misc/key.pem",
      .cert_file_name = "../misc/cert.pem",
      .passphrase = "1234"
        }).get("/*", [](auto *res, auto *req) {
        res->end("Hello world!");
    })

1>------ Build started: Project: ConsoleApplication1, Configuration: Debug Win32 ------
1>ConsoleApplication1.cpp
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(8): error C2059: syntax error: '.'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(7): error C2059: syntax error: ')'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(11): error C2059: syntax error: '.'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(7): error C2143: syntax error: missing ';' before '{'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(7): error C2143: syntax error: missing ')' before ';'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(7): error C2447: '{': missing function header (old-style formal list?)
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(13): error C2059: syntax error: ')'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(13): error C2059: syntax error: '.'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(13): error C2143: syntax error: missing ';' before '{'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(13): error C2143: syntax error: missing ')' before ';'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(13): error C2447: '{': missing function header (old-style formal list?)
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(17): error C2059: syntax error: ')'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(17): error C2059: syntax error: '.'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(19): error C2143: syntax error: missing ';' before '<<'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(19): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(19): error C2371: 'std::cout': redefinition; different basic types
1>c:\program files (x86)\microsoft visual studio\2017\enterprise\vc\tools\msvc\14.16.27023\include\iostream(29): note: see declaration of 'std::cout'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(20): error C2059: syntax error: '}'
1>c:\users\nova001845\source\repos\consoleapplication1\consoleapplication1\consoleapplication1.cpp(20): error C2143: syntax error: missing ';' before '}'
1>Done building project "ConsoleApplication1.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

**Comment by zhlongfj** [comment]
I built uwebsockets with vcpkg：vcpkg install uwebsockets

**Comment by ghost** [comment]
You need to enable C++17, this project is using modern C++ features

**Comment by zhlongfj** [comment]
I got that errors while I compiled it with C++17.(ISO C++17 Standard (/std:c++17))

**Comment by ddevienne** [comment, marked as answer]
Using the { .field = ... } notation is a C++20 feature, not C++17...

If you could please stick to pure C++17 (which I'm just about to be able to use at work...),
that would be great Alex. My $0.02... --DD

PS: See https://en.cppreference.com/w/cpp/language/aggregate_initialization
PPS: C++20 is not even officially out yet.

**Comment by ghost** [comment]
Ah, right. But you don't have to use it, you can construct that structure in old style and pass it in.
